File: spark_scala/src/cn/spark/study/core/AccumulatorTest.scala
package cn.spark.study.core

import java.util

import org.apache.spark.{Accumulator, SparkConf, SparkContext}

object AccumulatorTest {
  def main(args: Array[String]): Unit = {
    val conf: SparkConf = new SparkConf().setMaster("local").setAppName("accimulator")
    val sc = new SparkContext(conf)
    val sum = sc.accumulator(0)
    val numberList= Array("1","2","3","4","5")
    val numbers = sc.parallelize(numberList)
    numbers.foreach(number => println(number))
  }
}

File: study/src/main/java/cn/spark/study/core/AccumulatorTest.java
package cn.spark.study.core;

import org.apache.spark.Accumulator;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;

import java.util.Arrays;
import java.util.List;

/**
 * @ClassName Accumulator
 * @Deseription TODO
 * @Author lxy_m
 * @Date 2019/8/22 17:23
 * @Version 1.0
 */
public class AccumulatorTest {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setMaster("local").setAppName("accimulator");
        JavaSparkContext sc = new JavaSparkContext(conf);
        final Accumulator<Integer> sum = sc.accumulator(0);
        List<Integer> numberList = Arrays.asList(1, 2, 3, 4, 5);
        JavaRDD<Integer> numbers = sc.parallelize(numberList);
        numbers.foreach(new VoidFunction<Integer>() {
            @Override
            public void call(Integer v) throws Exception {
                sum.add(v);
            }
        });
        System.out.println(sum.value());
        sc.close();

    }


}
